### ISSUE TITLE
send udp directly from music player w/o microphone

### ISSUE THREAD

**Issue by alex1047378** [scottlawsonbc/audio-reactive-led-strip#210]
is it possible to send udp data directly from the player to the nodemcu without the need of a microphone?

Advantages: less noise, more accurate, less delay

i play music with kodi. following possibilities come to mind:
- kodi plugin / visualization: didn't find any useful information how to write visualizations or plugins which receive audio input for kodi
- pulseaudio fork output to some kind of processor which the sends udp packages

**Comment by joeybab3** [comment]
Scott is in the process of a rewrite that will allow nodejs plugins to be released at a future date. In the mean time what platform are you on?

**Comment by aamarioneta** [comment]
Ubuntu 14.04, will probably upgrade soon.

-----Original Message-----

**Comment by joeybab3** [comment]
There’s a few issues on here similar to yours I believe, you’re trying to get the audio playing on your device to stream directly to the arduino, rather than sampling from the mic and sending correct? If so I know on windows you can use the stereo mixer and I believe we figured out a solution for Linux previously I’ll have a look at the previous issues and see if I can find it.

**Comment by aamarioneta** [comment]
Yes, thats exactly what i want. However the stream has to be in format [i,r,g,b] if i understand the .ino code correctly. It also has to be calculated for the exact number of pixels. That means the raw audio stream has to be processed before it is streamed. Otherwise the processing has to be done on the esp8266. I dont know. Is it powerful enough?

-----Original Message-----

**Comment by joeybab3** [comment]
I think some of the fft would be a bit to much for something like the esp8266 to handle, so the processing would have to still be done on the computer but implementing loopback does decrease the delay and increase the overall quality of the animation in my experience
